### PR TITLE
Rfb rust keywords 3195 v4

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/7178

Describe changes:
- rfb: move keywords to pure rust
- rfb: move app-layer to pure rust as well, and convert unit test to SV one

Coming after the MQTT merged PR https://github.com/OISF/suricata/pull/11502, and before other protocols

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1986

https://github.com/OISF/suricata/pull/11560 with needed rebase to get green CI

cc @satta

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
  

Pipeline 21754

**Comment by victorjulien** [comment]
@satta are you able to review and/or test this?

**Comment by satta** [comment]
> @satta are you able to review and/or test this?

I'll take a look and test a bit.

**Review by jufajardini — COMMENTED** [review]
Docs change looks good to me :) Thanks for adding the dedicated ticket!

File: rust/src/rfb/logger.rs [hunk 1]
```diff
@@ -67,15 +69,14 @@ fn log_rfb(tx: &RFBTransaction, js: &mut JsonBuilder) -> Result<(), JsonError> {
         _ => (),
     }
     if let Some(security_result) = &tx.tc_security_result {
-        let _ = match security_result.status {
-            0 => js.set_string("security_result", "OK")?,
-            1 => js.set_string("security-result", "FAIL")?,
```
Comment (satta) [inline rust/src/rfb/logger.rs:72]: Just noticed, this one would have been different (`-` instead of `_`) so this actually fixed a potential bug!

**Comment by satta** [comment]
Looks good. I like how much more concise everything gets in Rust.
Is the minimum progress constraint gone now that previously was required in `DetectAppLayer*Register()`? Looks like `DetectHelper*Register()` now pins it to 0?

File: rust/src/rfb/logger.rs (hunk 1 above)
Comment (victorjulien) [reply rust/src/rfb/logger.rs:72]: Is this a bug in 7.0.x?

Comment (satta) [reply rust/src/rfb/logger.rs:72]: Rather an inconsistency, but yes. The current code in 7.0.x uses a slightly different field name for `FAIL`.

Comment (catenacyber) [reply rust/src/rfb/logger.rs:72]: Nice accidental fix :-)

https://redmine.openinfosecfoundation.org/issues/7198 for the bug in 7

**Comment by catenacyber** [comment]
> Is the minimum progress constraint gone now that previously was required in DetectAppLayer*Register()? Looks like DetectHelper*Register() now pins it to 0?

It is not gone.
Did I mess something here ? (I do not see an enum value like `HTTP2StateOpen` here)

The basic helper function means to handle the simple case.

**Comment by satta** [comment]
> > Is the minimum progress constraint gone now that previously was required in DetectAppLayer_Register()? Looks like DetectHelper_Register() now pins it to 0?
> 
> It is not gone. Did I mess something here ?

Nope, I don't think so.

> The basic helper function means to handle the simple case.

Yes, that's just what I meant. Previously, for example, for `rfb.name` we used:
```C
    DetectAppLayerMpmRegister(BUFFER_NAME, SIG_FLAG_TOCLIENT, 1, PrefilterGenericMpmRegister,
            GetData, ALPROTO_RFB, 1);
```
which has now been replaced by
```rust
    G_RFB_NAME_BUFFER_ID = DetectHelperBufferMpmRegister(
        b"rfb.name\0".as_ptr() as *const libc::c_char,
        b"rfb name\0".as_ptr() as *const libc::c_char,
        ALPROTO_RFB,
        true, //toclient
        false,
        rfb_name_get,
    );
```
which is basically (in `DetectHelperBufferMpmRegister`) a call to
```c
     DetectAppLayerMpmRegister(
                name, SIG_FLAG_TOCLIENT, 2, PrefilterGenericMpmRegister, GetData, alproto, 0);
```

so whether it's 1 or 0 is no longer controllable when using the helper. But I guess that's the basic case you referred to.

File: rust/src/rfb/logger.rs (hunk 1 above)
Comment (satta) [reply rust/src/rfb/logger.rs:72]: I can provide a patch for 7.0.7

Comment (satta) [reply rust/src/rfb/logger.rs:72]: https://github.com/OISF/suricata/pull/11606 if you want

**Comment by victorjulien** [comment]
Merged in #11616, thanks!

File: rust/src/rfb/logger.rs (hunk 1 above)
Comment (jasonish) [reply rust/src/rfb/logger.rs:72]: @catenacyber @satta From the comments here, was the field name changed in master as well? If so, I think we need an upgrade note.

Comment (catenacyber) [reply rust/src/rfb/logger.rs:72]: The field name was inconsistent depending on the value, most times with underscore, once with a dash, this last case was fixed

**Pull request closed without merging**